core/trie.go

```diff
@@ -0,0 +1,351 @@
+package core
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/bits-and-blooms/bitset"
+)
+
+type (
+	StoragePath  = bitset.BitSet
+	StorageValue = TrieNode
+)
+
+type (
+	TrieKey   = felt.Felt
+	TrieValue = felt.Felt
+)
```

Comment:
I personally prefer using the underlying types directly. IMHO, type aliasing usually creates an unnecessary layer of indirection. Using type names to describe "structure" and variable names to describe "purpose" is my preferred convention.

In the end this probably comes down to taste and I'm open to keeping things as they are.